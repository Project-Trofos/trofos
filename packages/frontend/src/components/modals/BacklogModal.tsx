import { Avatar, Button, Col, Form, Input, InputNumber, Modal, Row, Select } from 'antd';
import { UserOutlined } from '@ant-design/icons';
import React, { useState } from 'react';
import './BacklogModal.css';
import { useParams } from 'react-router-dom';
import { useAddBacklogMutation } from '../../api/backlog';
import type { BacklogSelect, BacklogFormFields } from './types/BacklogModal.types';

const { Option } = Select;
const { TextArea } = Input;

function BacklogModal(): JSX.Element {
  // These constants will most likely be passed down as props or
  // fetched from an API. Currently hardcoded for developement.
  const TYPES: BacklogSelect[] = [
    { id: 'story', name: 'Story' },
    { id: 'task', name: 'Task' },
    { id: 'bug', name: 'Bug' },
  ];
  const PRIORITIES: BacklogSelect[] = [
    { id: 'very_high', name: 'Very High' },
    { id: 'high', name: 'High' },
    { id: 'medium', name: 'Medium' },
    { id: 'low', name: 'Low' },
    { id: 'very_low', name: 'Very Low' },
  ];
  const SPRINTS = [{ id: 1, name: 'Sprint 1' }];
  const USERS = [
    { id: 901, name: 'User1' },
    { id: 902, name: 'User2' },
  ];

  const params = useParams();
  const [form] = Form.useForm();

  const [addBacklog] = useAddBacklogMutation();

  const [isModalVisible, setIsModalVisible] = useState(false);
  const [isLoading, setIsLoading] = useState(false);

  const showModal = () => {
    setIsModalVisible(true);
  };

  const handleOk = () => {
    setIsModalVisible(false);
  };

  const handleCancel = () => {
    setIsModalVisible(false);
  };

  const handleFormSubmit = async (data: FormData): Promise<void> => {
    setIsLoading(true);

    // for development
    const payload: BacklogFormFields = {
      ...data,
      projectId: Number(params.projectId),
    };
    try {
      await addBacklog(payload).unwrap();
      setIsModalVisible(false);
      form.resetFields();
      console.log('Success');
    } catch (e) {
      console.error(e);
    } finally {
      setIsLoading(false);
    }
  };

  const renderFooter = (): JSX.Element[] => [
    <Button form="newBacklog" key="submit" type="primary" htmlType="submit" loading={isLoading}>
      Create
    </Button>,
  ];

  const renderTypeSelect = (): JSX.Element => (
    <Form.Item name="type" label="Type" rules={[{ required: true }]}>
      <Select className="type-select" placeholder="Type of backlog">
        {TYPES.map((type) => (
          <Option key={type.id} value={type.id}>
            {type.name}
          </Option>
        ))}
      </Select>
    </Form.Item>
  );

  const renderSprintSelect = (): JSX.Element => (
    <Form.Item name="sprintId" label="Sprint">
      <Select className="sprint-select" placeholder="Select Sprint" allowClear>
        {SPRINTS.map((sprint) => (
          <Option key={sprint.id} value={sprint.id}>
            {sprint.name}
          </Option>
        ))}
      </Select>
    </Form.Item>
  );

  const renderPrioritySelect = (): JSX.Element => (
    <Form.Item name="priority" label="Priority">
      <Select className="priority-select" placeholder="Select Priority" allowClear>
        {PRIORITIES.map((priority) => (
          <Option key={priority.id} value={priority.id}>
            {priority.name}
          </Option>
        ))}
      </Select>
    </Form.Item>
  );

  const renderReporterSelect = (): JSX.Element => (
    <Form.Item name="reporterId" label="Reporter" rules={[{ required: true }]}>
      <Select className="reporter-select">
        {USERS.map((user) => (
          <Option key={user.id} value={user.id}>
            <Avatar className="reporter-avatar" style={{ backgroundColor: '#85041C' }} icon={<UserOutlined />} />
            <span className="select-username-text">{user.name}</span>
          </Option>
        ))}
      </Select>
    </Form.Item>
  );

  const renderAssigneeSelect = (): JSX.Element => (
    <Form.Item name="assigneeId" label="Assignee">
      <Select className="assignee-select" allowClear>
        {USERS.map((user) => (
          <Option key={user.id} value={user.id}>
            <Avatar className="assignee-avatar" style={{ backgroundColor: '#ccc' }} icon={<UserOutlined />} />
            <span className="select-username-text">{user.name}</span>
          </Option>
        ))}
      </Select>
    </Form.Item>
  );

  const renderContent = (): JSX.Element => (
    <Form id="newBacklog" form={form} onFinish={handleFormSubmit}>
      <Form.Item name="summary" rules={[{ required: true }]}>
        <Input className="summary-input" placeholder="* Type summary here..." />
      </Form.Item>
      {renderTypeSelect()}
      {renderSprintSelect()}
      {renderPrioritySelect()}
      <Row>
        <Col span={12}>{renderReporterSelect()}</Col>
        <Col span={12}>{renderAssigneeSelect()}</Col>
      </Row>
      <Form.Item name="points" label="Points">
        <InputNumber className="points-input" min={1} />
      </Form.Item>
      <Form.Item name="description">
        <TextArea className="description-textarea" placeholder="Description..." autoSize={{ minRows: 5, maxRows: 8 }} />
      </Form.Item>
    </Form>
  );

  return (
    <>
      <Button className="new-backlog-btn" type="primary" onClick={showModal}>
        New Backlog
      </Button>
      <Modal
        title="New Backlog"
        visible={isModalVisible}
        onOk={handleOk}
        onCancel={handleCancel}
        footer={renderFooter()}
        width={600}
      >
        {renderContent()}
      </Modal>
    </>
  );
}

export default BacklogModal;
