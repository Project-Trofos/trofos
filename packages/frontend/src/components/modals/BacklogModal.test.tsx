import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import BacklogModal from './BacklogModal';

describe('BacklogModal tests', () => {
  it('renders new backlog modal with correct fields', () => {
    const mockUpdateBacklogs = jest.fn();
    const { baseElement } = render(<BacklogModal updateBacklogs={mockUpdateBacklogs} />);
  
    // Open modal
    const button = screen.getByText(/New Backlog/i);
    fireEvent.click(button);
  
    // Ensure fields are present
    expect(screen.getByPlaceholderText('* Type summary here...')).toBeInTheDocument();
    expect(screen.getByLabelText('Type')).toBeInTheDocument();
    expect(screen.getByLabelText('Sprint')).toBeInTheDocument();
    expect(screen.getByLabelText('Priority')).toBeInTheDocument();
    expect(screen.getByLabelText('Reporter')).toBeInTheDocument();
    expect(screen.getByLabelText('Assignee')).toBeInTheDocument();
    expect(screen.getByLabelText('Points')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Description...')).toBeInTheDocument();
  
    // Compare with snapshot to ensure structure remains the same
    expect(baseElement).toMatchSnapshot();
  });
});
