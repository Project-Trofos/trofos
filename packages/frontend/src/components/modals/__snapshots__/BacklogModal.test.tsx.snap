// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BacklogModal tests renders new backlog modal with correct fields 1`] = `
<body
  class="ant-scrolling-effect"
  style="overflow: hidden; overflow-x: hidden; overflow-y: hidden;"
>
  <div>
    <button
      ant-click-animating-without-extra-node="false"
      class="ant-btn ant-btn-primary new-backlog-btn"
      type="button"
    >
      <span>
        New Backlog
      </span>
    </button>
  </div>
  <div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal"
          role="dialog"
          style="width: 600px;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <button
              aria-label="Close"
              class="ant-modal-close"
              type="button"
            >
              <span
                class="ant-modal-close-x"
              >
                <span
                  aria-label="close"
                  class="anticon anticon-close ant-modal-close-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="test-id"
              >
                New Backlog
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <form
                class="ant-form ant-form-horizontal"
                id="newBacklog"
              >
                <div
                  class="ant-form-item"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <input
                            class="ant-input summary-input"
                            id="summary"
                            placeholder="* Type summary here..."
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-form-item"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class="ant-form-item-required"
                        for="type"
                        title="Type"
                      >
                        Type
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="ant-select ant-select-in-form-item type-select ant-select-single ant-select-show-arrow"
                          >
                            <div
                              class="ant-select-selector"
                            >
                              <span
                                class="ant-select-selection-search"
                              >
                                <input
                                  aria-activedescendant="type_list_0"
                                  aria-autocomplete="list"
                                  aria-controls="type_list"
                                  aria-haspopup="listbox"
                                  aria-owns="type_list"
                                  autocomplete="off"
                                  class="ant-select-selection-search-input"
                                  id="type"
                                  readonly=""
                                  role="combobox"
                                  style="opacity: 0;"
                                  type="search"
                                  unselectable="on"
                                  value=""
                                />
                              </span>
                              <span
                                class="ant-select-selection-placeholder"
                              >
                                Type of backlog
                              </span>
                            </div>
                            <span
                              aria-hidden="true"
                              class="ant-select-arrow"
                              style="user-select: none;"
                              unselectable="on"
                            >
                              <span
                                aria-label="down"
                                class="anticon anticon-down ant-select-suffix"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-form-item"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class=""
                        for="sprintId"
                        title="Sprint"
                      >
                        Sprint
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="ant-select ant-select-in-form-item sprint-select ant-select-single ant-select-allow-clear ant-select-show-arrow"
                          >
                            <div
                              class="ant-select-selector"
                            >
                              <span
                                class="ant-select-selection-search"
                              >
                                <input
                                  aria-activedescendant="sprintId_list_0"
                                  aria-autocomplete="list"
                                  aria-controls="sprintId_list"
                                  aria-haspopup="listbox"
                                  aria-owns="sprintId_list"
                                  autocomplete="off"
                                  class="ant-select-selection-search-input"
                                  id="sprintId"
                                  readonly=""
                                  role="combobox"
                                  style="opacity: 0;"
                                  type="search"
                                  unselectable="on"
                                  value=""
                                />
                              </span>
                              <span
                                class="ant-select-selection-placeholder"
                              >
                                Select Sprint
                              </span>
                            </div>
                            <span
                              aria-hidden="true"
                              class="ant-select-arrow"
                              style="user-select: none;"
                              unselectable="on"
                            >
                              <span
                                aria-label="down"
                                class="anticon anticon-down ant-select-suffix"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-form-item"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class=""
                        for="priority"
                        title="Priority"
                      >
                        Priority
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="ant-select ant-select-in-form-item priority-select ant-select-single ant-select-allow-clear ant-select-show-arrow"
                          >
                            <div
                              class="ant-select-selector"
                            >
                              <span
                                class="ant-select-selection-search"
                              >
                                <input
                                  aria-activedescendant="priority_list_0"
                                  aria-autocomplete="list"
                                  aria-controls="priority_list"
                                  aria-haspopup="listbox"
                                  aria-owns="priority_list"
                                  autocomplete="off"
                                  class="ant-select-selection-search-input"
                                  id="priority"
                                  readonly=""
                                  role="combobox"
                                  style="opacity: 0;"
                                  type="search"
                                  unselectable="on"
                                  value=""
                                />
                              </span>
                              <span
                                class="ant-select-selection-placeholder"
                              >
                                Select Priority
                              </span>
                            </div>
                            <span
                              aria-hidden="true"
                              class="ant-select-arrow"
                              style="user-select: none;"
                              unselectable="on"
                            >
                              <span
                                aria-label="down"
                                class="anticon anticon-down ant-select-suffix"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row"
                >
                  <div
                    class="ant-col ant-col-12"
                  >
                    <div
                      class="ant-form-item"
                    >
                      <div
                        class="ant-row ant-form-item-row"
                      >
                        <div
                          class="ant-col ant-form-item-label"
                        >
                          <label
                            class="ant-form-item-required"
                            for="reporterId"
                            title="Reporter"
                          >
                            Reporter
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <div
                                class="ant-select ant-select-in-form-item reporter-select ant-select-single ant-select-show-arrow"
                              >
                                <div
                                  class="ant-select-selector"
                                >
                                  <span
                                    class="ant-select-selection-search"
                                  >
                                    <input
                                      aria-activedescendant="reporterId_list_0"
                                      aria-autocomplete="list"
                                      aria-controls="reporterId_list"
                                      aria-haspopup="listbox"
                                      aria-owns="reporterId_list"
                                      autocomplete="off"
                                      class="ant-select-selection-search-input"
                                      id="reporterId"
                                      readonly=""
                                      role="combobox"
                                      style="opacity: 0;"
                                      type="search"
                                      unselectable="on"
                                      value=""
                                    />
                                  </span>
                                  <span
                                    class="ant-select-selection-placeholder"
                                  />
                                </div>
                                <span
                                  aria-hidden="true"
                                  class="ant-select-arrow"
                                  style="user-select: none;"
                                  unselectable="on"
                                >
                                  <span
                                    aria-label="down"
                                    class="anticon anticon-down ant-select-suffix"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="down"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-col ant-col-12"
                  >
                    <div
                      class="ant-form-item"
                    >
                      <div
                        class="ant-row ant-form-item-row"
                      >
                        <div
                          class="ant-col ant-form-item-label"
                        >
                          <label
                            class=""
                            for="assigneeId"
                            title="Assignee"
                          >
                            Assignee
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <div
                                class="ant-select ant-select-in-form-item assignee-select ant-select-single ant-select-allow-clear ant-select-show-arrow"
                              >
                                <div
                                  class="ant-select-selector"
                                >
                                  <span
                                    class="ant-select-selection-search"
                                  >
                                    <input
                                      aria-activedescendant="assigneeId_list_0"
                                      aria-autocomplete="list"
                                      aria-controls="assigneeId_list"
                                      aria-haspopup="listbox"
                                      aria-owns="assigneeId_list"
                                      autocomplete="off"
                                      class="ant-select-selection-search-input"
                                      id="assigneeId"
                                      readonly=""
                                      role="combobox"
                                      style="opacity: 0;"
                                      type="search"
                                      unselectable="on"
                                      value=""
                                    />
                                  </span>
                                  <span
                                    class="ant-select-selection-placeholder"
                                  />
                                </div>
                                <span
                                  aria-hidden="true"
                                  class="ant-select-arrow"
                                  style="user-select: none;"
                                  unselectable="on"
                                >
                                  <span
                                    aria-label="down"
                                    class="anticon anticon-down ant-select-suffix"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="down"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-form-item"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class=""
                        for="points"
                        title="Points"
                      >
                        Points
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="ant-input-number ant-input-number-in-form-item points-input"
                          >
                            <div
                              class="ant-input-number-handler-wrap"
                            >
                              <span
                                aria-disabled="false"
                                aria-label="Increase Value"
                                class="ant-input-number-handler ant-input-number-handler-up"
                                role="button"
                                unselectable="on"
                              >
                                <span
                                  aria-label="up"
                                  class="anticon anticon-up ant-input-number-handler-up-inner"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="up"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                                    />
                                  </svg>
                                </span>
                              </span>
                              <span
                                aria-disabled="false"
                                aria-label="Decrease Value"
                                class="ant-input-number-handler ant-input-number-handler-down"
                                role="button"
                                unselectable="on"
                              >
                                <span
                                  aria-label="down"
                                  class="anticon anticon-down ant-input-number-handler-down-inner"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                            <div
                              class="ant-input-number-input-wrap"
                            >
                              <input
                                aria-valuemin="1"
                                autocomplete="off"
                                class="ant-input-number-input"
                                id="points"
                                role="spinbutton"
                                step="1"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-form-item"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <textarea
                            class="ant-input description-textarea"
                            id="description"
                            placeholder="Description..."
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn ant-btn-primary"
                form="newBacklog"
                type="submit"
              >
                <span>
                  Create
                </span>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
